### ISSUE TITLE
K8SSAND-1135 ⁃ Add support for authentication in Reaper web UI and REST API

### ISSUE THREAD

**Issue by adutra** [k8ssandra/k8ssandra-operator#276]
This is a follow-up ticket after #170.

Reaper REST API and web UI are currently always unauthenticated. This ticket is about adding authentication to them.

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-1135) by [Unito](https://www.unito.io)
┆epic: Reaper Controller Integration
┆fixVersions: k8ssandra-operator-v1.0.0
┆friendlyId: K8SSAND-1135
┆priority: Medium
